src/extension_commands.ml

```diff
@@ -534,6 +534,180 @@ module Copy_type_under_cursor = struct
     command Extension_consts.Commands.copy_type_under_cursor handler
 end
 
+module Search_by_type = struct
+  let extension_name = "Search By Type"
+
+  let ocaml_lsp_doesnt_support_search_by_type ocaml_lsp =
+    not (Ocaml_lsp.can_handle_search_by_type ocaml_lsp)
+
+  let rec _remove_duplicates ~cmp = function
+    | a :: (b :: _ as t) ->
+      if cmp a b then _remove_duplicates ~cmp t
+      else a :: _remove_duplicates ~cmp t
+    | remaining -> remaining
```

Comment:
Ah, I must have missed it the first time I looked, but there is a utility to do that in the `List` module (that comes from `Base`): `List.remove_consecutive_duplicates`